libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/RestEntitlementsCheckAction.java

```diff
@@ -242,6 +251,17 @@ private static Stream<Entry<String, CheckAction>> getTestEntries(Class<?> action
         return entries.stream();
     }
 
+    private static CheckedConsumer<Environment, Exception> createConsumerForMethod(Method method) {
+        Class<?>[] parameters = method.getParameterTypes();
+        if (parameters.length == 0) {
+            return env -> method.invoke(null);
+        }
+        if (parameters.length == 1 && parameters[0].equals(Environment.class)) {
+            return env -> method.invoke(null, env);
+        }
```

Comment:
(If we ever acquire a second injectable parameter here, I'd probably change this to loop over the parameters, building the argument list based on the parameter types, much like a real DI framework would.)

Reply:
++
But for now all the "extra" info we use is in the environment (e.g. dirs).
But yes, if we need to add more, we can either do that or have "fixed" patterns (either arg-less, or all the additional args in the right order).

Reply:
I don't really see the value in forcing the args to be in a certain order. Like, what if I want to inject A and C? Do I need to declare B for no reason just so I have the right args in the right order? Also, why is `(A, C)` ok but `(C, A)` is not?

It will not surprise you that I'd prefer just to have DI. 😂